tools/cldr-apps/src/main/java/org/unicode/cldr/web/DataSection.java

```diff
@@ -1052,6 +1053,19 @@ public String toJSONString() throws JSONException {
                 jo.put("helpHtml", getHelpHTML());
                 jo.put("rdf", getRDFURI());
 
+                final PatternPlaceholders placeholders = PatternPlaceholders.getInstance();
+                jo.put("placeholderStatus", placeholders.getStatus(xpath).name());
+                Map<String, PatternPlaceholders.PlaceholderInfo> placeholderInfoMap = placeholders.get(xpath);
+                if (placeholderInfoMap != null && !placeholderInfoMap.isEmpty()) {
+                    JSONObject placeholderInfo = new JSONObject();
+                    for (final Map.Entry<String, PatternPlaceholders.PlaceholderInfo> e : placeholderInfoMap.entrySet()) {
+                        JSONObject subInfo = new JSONObject();
+                        subInfo.put("name", e.getValue().name);
+                        subInfo.put("example", e.getValue().example);
+                        placeholderInfo.put(e.getKey(), subInfo);
+                    }
+                    jo.put("placeholderInfo", placeholderInfo);
+                }
```

Comment:
could use a subroutine

Reply:
> the jax-rs version

?

Reply:
somewhere in api?

Reply:
https://github.com/unicode-org/cldr/pull/2012/files#diff-6f10ee486b924aa8fdfd98c3dede317f9c9812b33e0dc58b7ef12b4c4e343ca8R195-R197

Reply:
VoteAPIHelper

```java
        PatternPlaceholders placeholders = PatternPlaceholders.getInstance();
         row.placeholderStatus = placeholders.getStatus(xpath);
         row.placeholderInfo = placeholders.get(xpath);
```

jax-rs can serialize these by itself.